metaflow/plugins/argo/argo_workflows.py

```diff
@@ -1393,49 +1422,149 @@ def _exit_hook_templates(self):
         # TODO: Add details to slack message
         templates = []
         if self.notify_on_error:
-            templates.append(
-                Template("notify-on-error").http(
-                    Http("POST")
-                    .url(self.notify_slack_webhook_url)
-                    .body(
-                        json.dumps(
-                            {
-                                "text": ":rotating_light: _%s/argo-{{workflow.name}}_ failed!"
-                                % self.flow.name
-                            }
-                        )
-                    )
-                )
-            )
+            templates.append(self._slack_error_template)
+            templates.append(self._pagerduty_alert_template)
         if self.notify_on_success:
-            templates.append(
-                Template("notify-on-success").http(
-                    Http("POST")
-                    .url(self.notify_slack_webhook_url)
-                    .body(
-                        json.dumps(
-                            {
-                                "text": ":white_check_mark: _%s/argo-{{workflow.name}}_ succeeded!"
-                                % self.flow.name
-                            }
-                        )
-                    )
-                )
-            )
+            templates.append(self._slack_success_template)
+            templates.append(self._pagerduty_change_template)
         if self.notify_on_error or self.notify_on_success:
             # Warning: terrible hack to workaround a bug in Argo Workflow where the
             #          templates listed above do not execute unless there is an
             #          explicit exit hook. as and when this bug is patched, we should
             #          remove this effectively no-op template.
+            # Note: We use the Http template because changing this to an actual no-op container had the side-effect of
+            # leaving LifecycleHooks in a pending state even when they have finished execution.
             templates.append(
                 Template("exit-hook-hack").http(
                     Http("GET")
-                    .url(self.notify_slack_webhook_url)
+                    .url(
+                        self.notify_slack_webhook_url
+                        or "https://events.pagerduty.com/v2/enqueue"
+                    )
                     .success_condition("true == true")
                 )
             )
         return templates
 
+    @property
```

Comment:
any reason why this needs to be a property?

Reply:
just a bit cleaner usage, but it does introduce unnecessary setters/delete to the object, so I can revert to a simple instance methods instead.